### DISCUSSION TITLE
How to generates a component without using Standalone Component?

### DISCUSSION THREAD

**Discussion by doggy8088** [cyrilletuzi/vscode-angular-schematics#923] [Q&A] [answered]
In the latest Angular Schematics seems can only generate component only in Standalone component. What if I am developing a previous version of Angular, how can I generate component using traditional NgModule support?

My current workaround is install previous version of Angular Schematics. Do you plan to support switch in the Angular Schematics that support both development model?

**Comment by cyrilletuzi** [comment, marked as answer]
Hello,

The switch already exists in the Pro edition. As stated in the documentation:

> While this extension is very popular, with nearly 1 million installations, it is *not* a tool developed by the Angular team or affiliated to Google in any way.
>
> It is months of *unpaid* work by a single contributor.
>
> **Maintaining legacy features has a cost. The cost companies making money do not invest to migrate should not be the charge of unpaid contributors. So legacy features are only available in the Pro edition.**
>
> So feel free to enjoy advanced features and support a contributor at the same time! 💖

**Reply by doggy8088** [reply]
Thank you!
